s5_base/run.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+# Copyright   2020 Reykjavik University (Author: Judy Fong - judyfong@ru.is)
+# Apache 2.0
+#
+# See ../README.txt for more info on data required.
+#SBATCH --output=logs/samrun%J.out
+#SBATCH --nodelist=terra
+
+. ./cmd.sh
+. ./path.sh
+# setup the steps and utils directories
+. ./setup.sh
+
+set -euo pipefail
+
+samromur_root=samromur
+num_jobs=50
+stage=0
+
+. utils/parse_options.sh
+
+#TODO: set the samromur data and metadata variables within path.sh
+mfccdir=`pwd`/mfccs
+num_threads=4
+num_jobs=10
+minibatch_size=128
+training_dataset_ratio=80
+
+if [ $stage -le 0 ]; then
+  echo "Create training data from samromur training"
+  local/make_samromur.sh $samromur_root training data
+  local/make_samromur.sh $samromur_root test data
+  local/make_samromur.sh $samromur_root eval data
+
+  utils/combine_data.sh data/train data/samromur_training
+  utils/combine_data.sh data/dev data/samromur_test
+  utils/combine_data.sh data/eval data/samromur_eval
+  utils/fix_data_dir.sh data/train
+  utils/fix_data_dir.sh data/dev
+  utils/fix_data_dir.sh data/eval
+fi
+
+# Prepare features
+if [ $stage -le 1 ]; then
+  echo "Make mfccs"
+  # Make MFCCs for each dataset
+  for name in train dev eval; do
+    steps/make_mfcc.sh --mfcc-config conf/mfcc.conf \
+      --nj ${num_jobs} --cmd "$train_cmd --max-jobs-run 99" \
+      data/${name} exp/make_mfcc/${name} $mfccdir
+    utils/fix_data_dir.sh data/${name}
+  done
+fi
+
+if [ $stage -le 2 ]; then
+  echo "Comute cmvn"
+  for name in train dev eval; do
+    steps/compute_cmvn_stats.sh data/${name} exp/make_mfcc/${name} $mfccdir
+    utils/fix_data_dir.sh data/${name}
+    utils/validate_data_dir.sh data/${name}
+  done
+fi
+
+if [ $stage -le 3 ]; then
+  echo "Prepare dict and lang directories"
+  local/prep_samromur_lang.sh data/train/text data
+fi
+
+if [ $stage -le 4 ]; then
+  echo "Create 3gram language model from kenlm"
+  cat data/train/text | cut -d' ' -f2- | sed -e \
+ 's/[,.?!]//g' > data/text_lm_training.txt
+
+  mkdir -p data/lang_3gram
+
+  # language  modeling files are typically in data/lang_Xgram
+  utils/slurm.pl --mem 16G logs/make_LM_3gsmall.log local/make_LM.sh --order 3 \
```

Comment:
the memory allocation should be in cmd

Reply:
And another similar fix is ofcourse use the num_jobs flag for --nj. But this is both something that can just be done later.

Reply:
I've moved memory to cmd but I'm told that utils/slurm.pl doesn't take in --nj so I need to keep num_jobs currently.